### PR TITLE
Address Misc Macbuild Tool Issues

### PR DESCRIPTION
**Pull request by dakanji**
Addresses a few issues flagged by ShellCheck in macbuild.tool.

ShellCheck Refs:
https://github.com/koalaman/shellcheck/wiki/SC2155
https://github.com/koalaman/shellcheck/wiki/SC2164
https://github.com/koalaman/shellcheck/wiki/SC2046
https://github.com/koalaman/shellcheck/wiki/SC2034

### CODE REVIEW

File: macbuild.tool [hunk 1]
```diff
@@ -6,7 +6,8 @@ package() {
     exit 1
   fi
 
-  local ver=$(cat Include/OpenCore.h | grep OPEN_CORE_VERSION | sed 's/.*"\(.*\)".*/\1/' | grep -E '^[0-9.]+$')
+  local ver
+  ver=$(cat Include/OpenCore.h | grep OPEN_CORE_VERSION | sed 's/.*"\(.*\)".*/\1/' | grep -E '^[0-9.]+$')
```
Comment (vit9696) [inline macbuild.tool:10]: Why is that needed?

File: macbuild.tool [hunk 2]
```diff
@@ -68,7 +69,7 @@ package() {
   popd || exit 1
 }
 
-cd $(dirname "$0")
+cd "$(dirname $0)" || exit 1
```
Comment (vit9696) [inline macbuild.tool:72]: This may not work if `$0` contains spaces as far as I know.

File: macbuild.tool [hunk 3]
```diff
@@ -78,6 +79,13 @@ DEPURLS=(
   'https://github.com/acidanthera/DuetPkg'
 )
 DEPBRANCHES=('master' 'master' 'master')
+
+export ARCHS
+export SELFPKG
+export DEPNAMES
+export DEPURLS
+export DEPBRANCHES
```
Comment (vit9696) [inline macbuild.tool:87]: Why is that needed?

File: macbuild.tool (hunk 1 above)
Comment (dakanji) [reply macbuild.tool:10]: macbuild.tool works perfectly well as it is ... run it dozens of times in last week.

So change not "Needed" ... just desirable in terms of maintaining standards.
I.E., no issue here but there is a potential issue that may happen elsewhere in development if such local variables are declared and assigned together.

See: https://github.com/koalaman/shellcheck/wiki/SC2155#problematic-code-in-the-case-of-local

File: macbuild.tool (hunk 2 above)
Comment (dakanji) [reply macbuild.tool:72]: Indeed.

On review, issue is actually that command expansion is not needed as not assigning output.

Will modify to `cd dirname "$0" || exit 1`

File: macbuild.tool (hunk 3 above)
Comment (dakanji) [reply macbuild.tool:87]: Basically same as local variables item.

The declared variables are not actually used anywhere in this script file but are for efibuild.sh which is run in a subshell.

While things work as it, technically should be exported in such instances. Again, something that is not an issue here but potentially elsewhere if not following such by default

Comment (vit9696) [reply macbuild.tool:87]: Hmm, does the tool unconditionally warn about these variables? If we cannot silence the warnings, I am fine to merge this, but in this case I request all the scripts to be checked with this tool on CI level, so that we do not violate these recommendations anywhere else in the future.

Could you update your PR and include such a check in `.travis.yml`?

Comment (dakanji) [reply macbuild.tool:87]: ShellCheck raises the error messages and warnings by default if that's what you mean.

In my case it is installed as a plugin in my editor (Atom).

I believe TravisCI comes with it by default but will need to read up on how to add.

I will squash the commits into one clean one and push. If I can figure the Travis thing out,  I will add it.

Otherwise will add an issue and look into dealing with it separately. 

PS. ShellCheck messages can be silenced by adding a line telling it to ignore a condition above where that condition appears.

This is needed as there can be reasons why something is done but you will obviously see the addition when reviewing.

**Comment by vit9696** [comment]
Ok, I merged this.
